### ISSUE TITLE
Chinese input is interrupted and disappears during the input of continuous long text

### ISSUE THREAD

**Issue by CodePlayer** [ianstormtaylor/slate#2795]
#### Do you want to request a _feature_ or report a _bug_?

bug

#### What's the current behavior?

![slate-chinese-error](https://user-images.githubusercontent.com/3793502/57899381-61530d00-788f-11e9-8e24-40f5b977d50e.gif)

When inputting Chinese, we need to input multiple Pinyin letters continuously, and then press the `Space` key to place the input text into the corresponding text field.

However, when using Slate, if you input long text continuously and have **not** pressed the `Space` key, the continuous input will be interrupted and will be cleared.

This bug is currently only available on the Chrome browser and will not appear on Firefox.

This bug can be reproduced on the official example (I am using the official example).

#### What's the expected behavior?

Continuous long text input will not be interrupted and will be retained correctly.

Related issue: #2349

**Comment by thesunny** [comment]
@ianstormtaylor I think this is related to the remove leaves update. I noticed that `set-text-from-dom-node` was not updated and is not working. Any code that uses that function is therefore also not working including composition and Android IME.

https://github.com/ianstormtaylor/slate/blob/master/packages/slate-react/src/utils/set-text-from-dom-node.js

I made a cursory look yesterday to fix it but am encountering bugs when doing `insertTextAtRange` which I'm not sure is me or also related bugs in the code since the update. It doesn't make sense to me why I'm getting the errors. This one would also be helpful to have fixed for me to finish Android work.
